src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +61,723 @@ const ActorInit En_Fz_InitVars = {
+    this->isFreezing = false;
+    this->isDespawning = true;
+    this->actor.flags &= ~1;
+    this->actionFunc = EnFz_Melt;
+    this->actor.speedXZ = 0.0f;
+    this->speedXZ = 0.0f;
+}
+
+void EnFz_Melt(EnFz* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.scale.y, 0.0006f, 0.0002f);
+    if (this->actor.scale.y < 0.006f) {
+        this->actor.scale.x += 0.0004f;
+        this->actor.scale.z += 0.0004f;
+    }
+
+    if (this->actor.scale.y < 0.004f) {
+        this->envAlpha -= 8;
+        if (this->envAlpha > 255) {
+            this->envAlpha = 0;
+        }
+    }
+
+    if (this->envAlpha == 0) {
+        EnFz_SetupDespawn(this, globalCtx);
+    }
+}
+
+void EnFz_SetupFreezingStationary(EnFz* this) {
+    this->state = 1;
+    this->timer = 40;
+    this->isAlwaysTrue = true;
+    this->isFreezing = true;
+    this->actor.flags |= 1;
+    this->actionFunc = EnFz_FreezingStationary;
+    this->actor.gravity = -1.0f;
+}
+
+void EnFz_FreezingStationary(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 isTimerMod8;
+    s16 primAlpha;
+
+    if (this->counter & 0xC0) {
+        EnFz_RotateTowardsLink(this);
+        EnFz_GetCollision(this, globalCtx);
+    } else {
+        isTimerMod8 = false;
+        primAlpha = 150;
+        func_8002F974(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
+        if ((this->counter & 0x3F) >= 48) {
+            primAlpha = 630 - ((this->counter & 0x3F) * 10);
+        }
+
+        accel.x = accel.z = 0.0f;
+        accel.y = 0.6f;
+        pos.x = this->actor.posRot.pos.x;
+        pos.y = this->actor.posRot.pos.y + 20.0f;
+        pos.z = this->actor.posRot.pos.z;
+        Matrix_RotateY((this->actor.shape.rot.y / (f32)0x8000) * M_PI, MTXMODE_NEW);
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f;
+        Matrix_MultVec3f(&vec1, &velocity);
+        if ((this->counter % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
+    s32 pad;
+    static EnFzSpawnIceSmokeFunc SpawnIceSmoke[] = {
+        EnFz_SpawnIceSmokeHiddenState,  // Hiding
+        EnFz_SpawnIceSmokeGrowingState, // Fully grown
+        EnFz_SpawnIceSmokeActiveState,  // Growing or Shrinking to/from hiding
+        EnFz_SpawnIceSmokeActiveState,  // melting from fire attack
+    };
+
+    this->counter++;
+    if (this->unusedTimer1) {
+        this->unusedTimer1--;
+    }
+    if (this->timer) {
+        this->timer--;
+    }
+    if (this->unusedTimer2) {
+        this->unusedTimer2--;
+    }
+
```

Comment:
```suggestion
    if (this->unusedTimer1 != 0) {
        this->unusedTimer1--;
    }
    if (this->timer != 0) {
        this->timer--;
    }
    if (this->unusedTimer2 != 0) {
        this->unusedTimer2--;
    }

```
Timers aren't booleans, so they need to be compared to 0. Assuming that doesn't break match.